src/TRestAxionOpticsMirror.cxx

```diff
@@ -0,0 +1,716 @@
+
+    fReflectivityTable.clear();
+    fTransmissionTable.clear();
+    map<Int_t, std::vector<Float_t>> reflectivity;
+    map<Int_t, std::vector<Float_t>> transmission;
+    for (double n = 0; n < 9; n = n + 4.5) {
+        fHenkeKeys["Min"] = DoubleToString(n);
+        fHenkeKeys["Max"] = DoubleToString(n + 4.5);
+
+        cout.flush();
+        vector<Float_t> reflect;
+        vector<Float_t> transm;
+        reflect.clear();
+        transm.clear();
+
+        cout << "Scanning angles between " << n << " and " << n + 4.5 << ".";
+        for (int e = 30; e <= 15000; e += 30) {
+            fHenkeKeys["Fixed"] = IntegerToString(e);
+            cout << ".";
+            cout.flush();
+            string fname = DownloadHenkeFile();
+
+            std::vector<std::vector<Double_t>> data;
+            TRestTools::ReadASCIITable(fname, data, 2);
+
+            // we skip the last point if we are not at the latest angles file
+            Int_t N = data.size() - 1;
+            if (n == 4.5) N = N + 1;
+
+            for (int m = 0; m < N; m++) reflectivity[e].push_back((Float_t)data[m][1]);
+            for (int m = 0; m < N; m++) transmission[e].push_back((Float_t)data[m][2]);
+        }
+        cout << endl;
+    }
+    cout << "Number of angles stored : " << reflectivity[30].size() << endl;
+
+    for (const auto& x : reflectivity) {
+        fReflectivityTable.push_back(x.second);
+    }
+    for (const auto& x : transmission) {
+        fTransmissionTable.push_back(x.second);
+    }
+
+    ExportTables();
+}
+
+///////////////////////////////////////////////
+/// \brief It returns the corresponding reflectivity filename for the mirror properties defined in the data
+/// members
+///
+std::string TRestAxionOpticsMirror::GetReflectivityFilename() {
+    string fnameR = "Reflectivy_" + fMirrorType + "_" + fLayerTop + "_" + fLayerThicknessTop + "_" +
+                    fSubstrate + "_" + fSigmaTop + ".N901f";
+    if (fMirrorType == "Bilayer")
+        fnameR = "Reflectivity_" + fMirrorType + "_" + fLayerTop + "_" + fLayerThicknessTop + "_" +
+                 fSigmaTop + "_" + fLayerBottom + "_" + fLayerThicknessBottom + "_" + fSigmaBottom + "_" +
+                 fSubstrate + ".N901f";
+    return fnameR;
+}
+
+///////////////////////////////////////////////
+/// \brief It returns the corresponding transmission filename for the mirror properties defined in the data
+/// members
+///
+std::string TRestAxionOpticsMirror::GetTransmissionFilename() {
+    string fnameT = "Transmission_" + fMirrorType + "_" + fLayerTop + "_" + fLayerThicknessTop + "_" +
+                    fSubstrate + "_" + fSigmaTop + ".N901f";
+    if (fMirrorType == "Bilayer")
+        fnameT = "Transmission_" + fMirrorType + "_" + fLayerTop + "_" + fLayerThicknessTop + "_" +
+                 fSigmaTop + "_" + fLayerBottom + "_" + fLayerThicknessBottom + "_" + fSigmaBottom + "_" +
+                 fSubstrate + ".N901f";
+    return fnameT;
+}
```

Comment:
The changes are appreciated, but it's sketchy to not only have hardcoded numbers appearing in the code (numbers without context or explanation are scary!), but also having the filename be technically independent of those numbers. If someone ever changes the number of data points, they will almost certainly forget to change the filename. 

So imo, there should at the very least be a private variable of the class that stores the number of points that is used both in the code (e.g. when reading the binary file) as well as to generate the correct file name. At the very least the read / save binary file calls should get an argument for number of points that is then used to generate the correct extension. That way at least future readers of the code know that this can change.

Reply:
We will never change the number of data points for this class.

Reply:
I thought so too for our version but then the hype about the 57Fe axion got bigger and we had to add 500 more data points to go up to 15keV.

Reply:
We will need a precision better than 0.01 degrees?

Reply:
If you wish, you could write an issue about, and do the changes in an independent PR

Reply:
I don't think we need a precision better than 0.01° as we don't know the mirror angles that precise anyway but it might be possible that someone wants to go up to higher energies in the future and keep the precision we have. But sure, we can make an issue about it.